File: python/example/definitions-recv.py
# Example Want Definitions Recv - defines how to use want for receiving actions
# ARBITER set action defs

# In Python you must import want to use as
# a decorator
from riffle import want

# After importing want, you simply decorate 
# any function with it.
@want(str)
def myFunction(aString):
    print(aString) # Guaranteed str
# resulting function will only be called with
# the args if they are of the type specified

# NOTE: each @want must decorate a function
# below, we removed them for clarity

# Nothing is returned
@want()

# The primitives
@want(str)
@want(int)
@want(float)
@want(bool)

# Collections
@want(list) # A list of anything
@want(dict) # A dict containing anything

# None is returned
@want(None)

# Many arguments
@want(str, int, float) # 3 args: str, int, float
@want(str, [int]) # 2 args: str, list of many int's

# End Example Want Definitions Recv

File: js/example/definitions-recv.js
// Example Want Definitions Recv - defines how to use want for receiving actions
// ARBITER set action defs

// In NodeJS you must import jsriffle:
var riffle = require('jsriffle');

// After importing riffle, simply add types to
// function declarations wrapped with riffle.want
this.Register("endpoint", riffle.want(function(s) {
    console.log(s);
}, String));
// the registered endpoint will only execute
// if it is provided a String

// Nothing is returned
riffle.want(function () { ... })

// The primitives
riffle.want(function (s) { ... }, String)
riffle.want(function (i) { ... }, Number)
riffle.want(function (f) { ... }, Number)
riffle.want(function (b) { ... }, Boolean)

// Collections
// A list of anything
riffle.want(function (l) { ... }, [])
// A dict of anything
riffle.want(function (d) { ... }, {})

// Many arguments
// This function requires a string and 2 numbers
riffle.want(function (s, i, f) { ... }, String, Number, Number)
// This function requires a string and a list of numbers
riffle.want(function (s, l) { ... }, String, [Number])

// End Example Want Definitions Recv
